contextualized/regression/datasets.py

```diff
@@ -135,3 +135,14 @@ def __init__(self, dataset):
 
     def __iter__(self):
         return iter(self.dataset)
+
+
+def singletask_worker_init_fn(worker_id):
+    worker_info = torch.utils.data.get_worker_info()
+    dataset = worker_info.dataset
+    split_size = len(dataset.dataset.C) // worker_info.num_workers  # divide equally among workers
+    start = worker_id*split_size
+    end = (worker_id+1)*split_size
+    dataset.dataset.C = dataset.dataset.C[start: end]
```

Comment:
I assume this is not a problem to over-write `dataset.dataset.C` with a slice of the dataset because this function is only called once (on worker init). Please add a comment or assertion about this expected behavior (i.e. this function should never be called twice from the same worker?).

Reply:
The worker function is called once for each worker. So over-writing dataset.dataset.C should not cause issues.

Reply:
The worker function is passed as a keyword argument to the pytroch DataLoader. Moreover, the pytorch back-end only allows the worker function to run once per worker. So over-writing dataset.dataset.C should not cause issues.